support/ppMemUsage.hpp

```diff
@@ -0,0 +1,23 @@
+#include <malloc.h> //warning - this is GNU-specific
+
+void hostGetMem(size_t* free, size_t* total) {
+  const double M = 1024*1024;
+#if defined(__GNUC__) && defined(PUMI_HAS_MALLINFO2)
```

Comment:
We'll need to rename `PUMI_HAS_MALLINFO2` to `PUMIPIC_HAS_MALLINFO2` and add the cmake logic here:

https://github.com/SCOREC/core/blob/8436909d489f6d99a199239ff835d82dd94381c2/CMakeLists.txt#L181-L189

I forgot about this additional check.  Sorry about that.

Reply:
Can you check my cmake logic? I am getting this result: Looking for mallinfo2 - not found

Reply:
That appears to be correct on cranium with gcc 10.  On my desktop with gcc 12 the cmake logic finds mallinfo2 and the `main.cpp` listed below compiles.

```
~/ThrowAwayCode/cmakeMallInfo $ which g++
/opt/scorec/spack/v0154_2/install/linux-rhel7-x86_64/gcc-6.5.0/gcc-10.1.0-tf5jjaditemasrbsl7tz6pnqa6duqwkg/bin/g++

~/ThrowAwayCode/cmakeMallInfo $ cat main.cpp 
#include <malloc.h> //warning - this is GNU-specific

void hostGetMem(size_t* free, size_t* total) {
  const double M = 1024*1024;
  struct mallinfo2 meminfo_now = mallinfo2();
  *total = meminfo_now.arena/M;
  *free = meminfo_now.fordblks/M;
}

void main() {
  size_t free, total;
  hostGetMem(&free, &total);
}

~/ThrowAwayCode/cmakeMallInfo $ g++ main.cpp -o main
main.cpp: In function 'void hostGetMem(size_t*, size_t*)':
main.cpp:5:20: error: variable 'hostGetMem(size_t*, size_t*)::mallinfo2 meminfo_now' has initializer but incomplete type
    5 |   struct mallinfo2 meminfo_now = mallinfo2();
      |                    ^~~~~~~~~~~
main.cpp:5:44: error: invalid use of incomplete type 'struct hostGetMem(size_t*, size_t*)::mallinfo2'
    5 |   struct mallinfo2 meminfo_now = mallinfo2();
      |                                            ^
main.cpp:5:10: note: forward declaration of 'struct hostGetMem(size_t*, size_t*)::mallinfo2'
    5 |   struct mallinfo2 meminfo_now = mallinfo2();
      |          ^~~~~~~~~
main.cpp: At global scope:
main.cpp:10:1: error: '::main' must return 'int'
   10 | void main() {
      | ^~~~
```

Reply:
I believe that message implies that mallinfo2 was not found. I get the same message when I run that code on my scrabble. when I run the code on my ubuntu vm it compiles.

Reply:
Correct.  I was just providing a simpler standalone code to check the logic on different systems.